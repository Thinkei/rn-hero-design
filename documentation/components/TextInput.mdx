---
name: TextInput
menu: Components
---

import { Playground } from 'docz'
import { TextInput } from 'rn-hero-design'
import { IPhoneX } from './DeviceFrame'

# TextInput

## Basic usage

<Playground>
  <IPhoneX screenStyle={{ padding: 16, paddingTop: 48 }}>
    <TextInput value="" label="Email" />
    <TextInput value="toan.nguyen@employmenthero.com" label="Email" rightIcon="email-outline" />
    <TextInput value="toan.nguyen$employmenthero.com" label="Email" rightIcon="email-outline" error="Invalid email" />
    <TextInput value="toan.nguyen$employmenthero.com" label="Email" rightIcon="email-outline" disabled />
  </IPhoneX>
</Playground>

## Properties

