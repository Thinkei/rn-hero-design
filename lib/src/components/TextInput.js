import React from 'react';
import t from 'prop-types';
import {
  View,
  Text,
  TextInput as RNTextInput,
  TouchableWithoutFeedback,
  StyleSheet,
} from 'react-native';
import Icon from './Icon';
import heroTheme from '../themes/hero';
import composeStyles from '../helpers/composeStyles';

class TextInput extends React.Component {
  state = {
    focused: false,
  };

  handleFocus = () => this.setState({ focused: true });

  handleBlur = () => this.setState({ focused: false });

  render() {
    const { focused } = this.state;
    const {
      label,
      value,
      onChangeText,
      onPressIcon,
      rightIcon,
      disabled,
      error,
      autoFocus,
      secureTextEntry,
      wrapperStyle = {},
      labelStyle = {},
      inputStyle = {},
      iconStyle = {},
      theme = heroTheme,
    } = this.props;

    return (
      <View style={StyleSheet.flatten([theme.textInput.wrapper, wrapperStyle])}>
        <Text
          style={StyleSheet.flatten([
            theme.textInput.label,
            labelStyle,
            error ? theme.textInput.errorLabel : null,
          ])}
        >
          {focused || value ? label : ''}
        </Text>
        <View
          style={StyleSheet.flatten([
            theme.textInput.textInput,
            inputStyle,
            error ? theme.textInput.errorTextInput : null,
          ])}
        >
          <RNTextInput
            placeholder={focused ? '' : label}
            value={value}
            onChangeText={onChangeText}
            onFocus={this.handleFocus}
            onBlur={this.handleBlur}
            editable={!disabled}
            autoFocus={autoFocus}
            secureTextEntry={secureTextEntry}
            placeholderTextColor={
              composeStyles([
                theme.textInput.baseTextInput,
                inputStyle,
                disabled ? theme.textInput.disabledBaseTextInput : null,
              ]).color
            }
            style={StyleSheet.flatten([
              theme.textInput.baseTextInput,
              inputStyle.color ? { color: inputStyle.color } : null,
              disabled ? theme.textInput.disabledBaseTextInput : null,
            ])}
          />
          <TouchableWithoutFeedback onPress={onPressIcon}>
            <Icon
              icon={rightIcon}
              color={
                composeStyles([
                  theme.textInput.icon,
                  iconStyle,
                  error ? theme.textInput.errorIcon : null,
                ]).color
              }
              wrapperStyle={theme.textInput.icon}
            />
          </TouchableWithoutFeedback>
        </View>
        <Text style={theme.textInput.errorMessage}>{error}</Text>
      </View>
    );
  }
}

TextInput.propTypes = {
  /**
   * The label and placeholder of text input
   */
  label: t.string.isRequired,
  value: t.string.isRequired,
  onChangeText: t.func.isRequired,
  onPressIcon: t.func,
  rightIcon: t.string,
  disabled: t.bool,
  error: t.string,
  autoFocus: t.bool,
  secureTextEntry: t.bool,
  wrapperStyle: t.object,
  labelStyle: t.object,
  inputStyle: t.object,
  iconStyle: t.object,
  /**
   * The theme object contains style of all components. Can be injected by `injectTheme` HOC
   */
  theme: t.object,
};

export default TextInput;
