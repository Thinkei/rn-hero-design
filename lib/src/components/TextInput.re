open ReactNative;

let isEmpty = str => String.length(str) == 0;

let emptyStyle = Style.style();

let noop = _ => ();

[@bs.get] external getColorProperty: Style.t => Color.t = "color";

[@react.component]
let make =
    (
      ~testID="",
      ~label="",
      ~placeholder="",
      ~value=?,
      ~onChange=noop,
      ~onChangeText=noop,
      ~onSelectionChange=noop,
      ~onKeyPress=noop,
      ~onFocus=noop,
      ~onBlur=noop,
      ~onPressIcon=noop,
      ~rightIcon="",
      ~disabled=false,
      ~error="",
      ~autoFocus=false,
      ~autoCorrect=false,
      ~secureTextEntry=false,
      ~multiline=false,
      ~wrapperStyle=emptyStyle,
      ~labelStyle=emptyStyle,
      ~inputStyle=emptyStyle,
      ~iconStyle=emptyStyle,
      ~errorStyle=emptyStyle,
      ~children=React.null,
      ~theme=Hero_Theme.default,
    ) => {
  let (focused, setFocused) = React.useState(() => false);
  let placeholder_ =
    switch (placeholder, focused) {
    | ("", true) => ""
    | ("", false) => label
    | _ => placeholder
    };
  let label_ =
    switch (value, focused) {
    | (_, true) => label
    | (Some(value), _) when !isEmpty(value) => label
    | _ => ""
    };

  let handleFocus =
    React.useCallback2(
      _ => {
        setFocused(_ => true);
        onFocus();
        ();
      },
      (onFocus, setFocused),
    );

  let handleBlur =
    React.useCallback2(
      _ => {
        setFocused(_ => false);
        onBlur();
        ();
      },
      (onBlur, setFocused),
    );

  <View
    style={StyleSheet.flatten([|theme##textInput##wrapper, wrapperStyle|])}>
    <Text
      style={StyleSheet.flatten([|
        theme##textInput##label,
        focused ? theme##textInput##activeLabel : emptyStyle,
        !isEmpty(error) ? theme##textInput##errorLabel : emptyStyle,
        labelStyle,
      |])}>
      label_->React.string
    </Text>
    <View
      style={StyleSheet.flatten([|
        theme##textInput##textInput,
        focused ? theme##textInput##activeTextInput : emptyStyle,
        !isEmpty(error) ? theme##textInput##errorTextInput : emptyStyle,
        inputStyle,
      |])}>
      <RNTextInput
        testID
        placeholder=placeholder_
        ?value
        onChange
        onChangeText
        onSelectionChange
        onKeyPress
        onFocus=handleFocus
        onBlur=handleBlur
        editable={!disabled}
        autoFocus
        autoCorrect
        secureTextEntry
        multiline
        scrollEnabled=false
        placeholderTextColor={
          StyleSheet.flatten([|
            theme##textInput##baseTextInput,
            inputStyle,
            disabled ? theme##textInput##disabledBaseTextInput : emptyStyle,
            isEmpty(placeholder) ? emptyStyle : theme##textInput##placeholder,
          |])
          ->getColorProperty
        }
        style={StyleSheet.flatten([|
          theme##textInput##baseTextInput,
          !
            Js.Nullable.isNullable(
              inputStyle->getColorProperty->Js.Nullable.return,
            )
            ? Style.style(~color=inputStyle->getColorProperty, ())
            : emptyStyle,
          disabled ? theme##textInput##disabledBaseTextInput : emptyStyle,
        |])}>
        children
      </RNTextInput>
      <TouchableWithoutFeedback onPress=onPressIcon>
        <Icon
          icon=rightIcon
          size=20.0
          color={
            StyleSheet.flatten([|
              theme##textInput##icon,
              focused ? theme##textInput##activeIcon : emptyStyle,
              !isEmpty(error) ? theme##textInput##errorIcon : emptyStyle,
              iconStyle,
            |])
            ->getColorProperty
          }
          wrapperStyle={theme##textInput##icon}
        />
      </TouchableWithoutFeedback>
    </View>
    <Text
      style={StyleSheet.flatten([|
        theme##textInput##errorMessage,
        errorStyle,
      |])}>
      error->React.string
    </Text>
  </View>;
};

let default = Helpers.injectTheme(make);
