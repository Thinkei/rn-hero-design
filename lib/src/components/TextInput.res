open ReactNative

let isEmpty = str => String.length(str) == 0

let emptyStyle = Style.style()

let noop = _ => ()

@deriving(jsConverter)
type keyboardType = [
  | #default
  | @as("number-pad") #numberPad
  | @as("decimal-pad") #decimalPad
  | #numeric
  | @as("email-address") #emailAddress
  | @as("phone-pad") #phonePad
  | @as("ascii-capable") #asciiCapable
  | @as("numbers-and-punctuation") #numbersAndPunctuation
  | #url
  | @as("name-phone-pad") #namePhonePad
  | #twitter
  | @as("web-search") #webSearch
  | @as("visible-password") #visiblePassword
]

@deriving(jsConverter)
type autoCapitalize = [#none | #sentences | #words | #characters]

@get external getColorProperty: Style.t => Color.t = "color"

@react.component
let make = (
  ~testID="",
  ~label="",
  ~placeholder="",
  ~keyboardType: option<string>=?,
  ~autoCapitalize: option<string>=?,
  ~defaultValue=?,
  ~value=?,
  ~onChange=noop,
  ~onChangeText=noop,
  ~onSelectionChange=noop,
  ~onKeyPress=noop,
  ~onFocus=noop,
  ~onBlur=noop,
  ~onPressIcon=noop,
  ~rightIcon="",
  ~disabled=false,
  ~error="",
  ~autoFocus=false,
  ~autoCorrect=false,
  ~secureTextEntry=false,
  ~multiline=false,
  ~wrapperStyle=emptyStyle,
  ~labelStyle=emptyStyle,
  ~inputStyle=emptyStyle,
  ~iconStyle=emptyStyle,
  ~errorStyle=emptyStyle,
  ~children=React.null,
  ~theme=Hero_Theme.default,
) => {
  let (focused, setFocused) = React.useState(() => false)
  let (internalValue, setInternalValue) = React.useState(() => defaultValue)

  let isUncontrolled = Belt.Option.isNone(value)
  let placeholder_ = switch (placeholder, focused) {
  | ("", true) => ""
  | ("", false) => label
  | _ => placeholder
  }
  /* defaultValue triggers re-render! So we need to handle showing label on defaultValue changed */
  let label_ = switch (value, internalValue, defaultValue, focused) {
  | (_, _, _, true) => label
  | (Some(value), _, _, _) if !isEmpty(value) => label
  | (None, Some(internalValue), _, _) if !isEmpty(internalValue) => label
  | (None, None, Some(defaultValue), _) if !isEmpty(defaultValue) => label
  | (None, Some(""), Some(defaultValue), _) if !isEmpty(defaultValue) => label
  | _ => ""
  }
  let keyboardType_ =
    keyboardType->Belt.Option.flatMap(keyboardTypeFromJs)->Belt.Option.getWithDefault(#default)

  let autoCapitalize_ = autoCapitalize->Belt.Option.flatMap(autoCapitalizeFromJs)

  let handleFocus = React.useCallback2(_ => {
    setFocused(_ => true)
    onFocus()
    ()
  }, (onFocus, setFocused))

  let handleBlur = React.useCallback2(_ => {
    setFocused(_ => false)
    onBlur()
    ()
  }, (onBlur, setFocused))

  let handleChangeText = React.useCallback2(text => {
    if isUncontrolled {
      setInternalValue(_ => Some(text))
    }
    onChangeText(text)
    ()
  }, (onChangeText, setInternalValue))

  <View style={StyleSheet.flatten([theme["textInput"]["wrapper"], wrapperStyle])}>
    <Text
      style={StyleSheet.flatten([
        theme["textInput"]["label"],
        focused ? theme["textInput"]["activeLabel"] : emptyStyle,
        !isEmpty(error) ? theme["textInput"]["errorLabel"] : emptyStyle,
        labelStyle,
      ])}>
      {label_->React.string}
    </Text>
    <View
      style={StyleSheet.flatten([
        theme["textInput"]["textInput"],
        focused ? theme["textInput"]["activeTextInput"] : emptyStyle,
        !isEmpty(error) ? theme["textInput"]["errorTextInput"] : emptyStyle,
        inputStyle,
      ])}>
      <ReactNative.TextInput
        testID
        placeholder=placeholder_
        keyboardType=keyboardType_
        autoCapitalize=?autoCapitalize_
        ?defaultValue
        ?value
        onChange
        onChangeText=handleChangeText
        onSelectionChange
        onKeyPress
        onFocus=handleFocus
        onBlur=handleBlur
        editable={!disabled}
        autoFocus
        autoCorrect
        secureTextEntry
        multiline
        scrollEnabled=false
        placeholderTextColor={theme["textInput"]["placeholder"]->getColorProperty}
        style={StyleSheet.flatten([
          theme["textInput"]["baseTextInput"],
          !Js.Nullable.isNullable(inputStyle->getColorProperty->Js.Nullable.return)
            ? Style.style(~color=inputStyle->getColorProperty, ())
            : emptyStyle,
          disabled ? theme["textInput"]["disabledBaseTextInput"] : emptyStyle,
        ])}>
        children
      </ReactNative.TextInput>
      <TouchableWithoutFeedback onPress=onPressIcon>
        <View>
          <Icon
            icon=rightIcon
            size=20.0
            color={StyleSheet.flatten([
              theme["textInput"]["icon"],
              focused ? theme["textInput"]["activeIcon"] : emptyStyle,
              !isEmpty(error) ? theme["textInput"]["errorIcon"] : emptyStyle,
              iconStyle,
            ])->getColorProperty}
            wrapperStyle={theme["textInput"]["icon"]}
          />
        </View>
      </TouchableWithoutFeedback>
    </View>
    <Text style={StyleSheet.flatten([theme["textInput"]["errorMessage"], errorStyle])}>
      {error->React.string}
    </Text>
  </View>
}

let default = Helpers.injectTheme(make)
