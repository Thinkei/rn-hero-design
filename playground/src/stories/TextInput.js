import React from 'react';
import { ScrollView } from 'react-native';
import {
  Text,
  TextInput,
  BottomButton,
  KeyboardAvoidingView,
  injectTheme,
} from 'rn-hero-design';

const noop = () => {};

const TextInputScreen = ({ theme }) => (
  <KeyboardAvoidingView withNavigation style={{ flex: 1 }}>
    <ScrollView
      contentContainerStyle={{ padding: theme.variables.MEDIUM_SIZE }}>
      <TextInput
        label="Email"
        defaultValue="gia.toan@employmenthero.com"
        rightIcon="email-outline"
        onChange={console.log}
      />

      <TextInput
        keyboardType="numeric"
        label="Password"
        value="12345678"
        rightIcon="eye-outline"
        onChangeText={noop}
        secureTextEntry
      />

      <TextInput
        keyboardType="numeric"
        label="Confirm password"
        value="1234567"
        rightIcon="eye-outline"
        onChangeText={noop}
        secureTextEntry
        error="Password not matched"
      />

      <TextInput label="Region" value="Vietnam" disabled onChangeText={noop} />

      <Text size="h5">* By sign-up, you agree to our terms and conditions</Text>
    </ScrollView>

    <BottomButton text="Register" onPress={() => {}} />
  </KeyboardAvoidingView>
);

export default injectTheme(TextInputScreen);
