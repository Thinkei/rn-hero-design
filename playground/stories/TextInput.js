import React from 'react';
import { View, SafeAreaView, StyleSheet } from 'react-native';
import { TextInput, Button, Text } from 'rn-hero-design';

const noop = () => {};

const TextInputScreen = () => (
  <SafeAreaView style={styles.safeArea}>
    <View style={styles.container}>
      <View style={styles.inputGroup}>
        <TextInput
          label="Email"
          value="toan.nguyen@employmenthero.com"
          rightIcon="email-outline"
          onChange={noop}
        />

        <TextInput
          label="Password"
          value="123456"
          rightIcon="eye-outline"
          onChange={noop}
          secureTextEntry
        />

        <TextInput
          label="Confirm password"
          value="12345"
          rightIcon="eye-outline"
          onChange={noop}
          secureTextEntry
          error="Password not matched"
          wrapperStyle={{ marginBottom: 32 }}
        />

        <TextInput label="Region" value="Vietnam" disabled onChange={noop} />

        <Text>* By signing up, you agree to our terms and conditions</Text>
      </View>

      <Button text="Register" onPress={() => {}} />
    </View>
  </SafeAreaView>
);

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
  },
  container: {
    flex: 1,
    paddingHorizontal: 16,
    paddingTop: 16,
  },
  inputGroup: {
    flex: 1,
  },
});

export default TextInputScreen;
