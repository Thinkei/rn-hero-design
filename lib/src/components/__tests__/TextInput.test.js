import React from 'react';
import TextInput from '../TextInput.gen';
import renderer from 'react-test-renderer';

describe('TextInput', () => {
  it.each`
    element
    ${<TextInput value="" onChangeText={() => {}} label="" />}
    ${<TextInput value="" onChangeText={() => {}} label="Email" />}
    ${<TextInput value="" onChangeText={() => {}} label="Email" rightIcon="email" />}
    ${<TextInput value="" onChangeText={() => {}} label="Email" rightIcon="email" disabled />}
    ${<TextInput value="" onChangeText={() => {}} label="Email" rightIcon="email" disabled error="error" />}
  `('renders correctly', ({ element }) => {
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it.each`
    element
    ${<TextInput value="kienhiepsi" keyboardType="name-phone-pad" />}
    ${<TextInput value="kienhiepsi" keyboardType="web-search" />}
    ${<TextInput value="kienhiepsi" keyboardType="number-pad" />}
    ${<TextInput value="kienhiepsi" keyboardType="decimal-pad" />}
    ${<TextInput value="kienhiepsi" keyboardType="phone-pad" />}
    ${<TextInput value="kienhiepsi" keyboardType="ascii-capable" />}
    ${<TextInput value="kienhiepsi" keyboardType="numbers-and-punctuation" />}
    ${<TextInput value="kienhiepsi" keyboardType="visible-password" />}
  `('renders keyboardType correctly', ({ element }) => {
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
