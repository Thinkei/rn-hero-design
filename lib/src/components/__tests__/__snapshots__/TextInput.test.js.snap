// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#c3c6cc",
        "fontSize": 13.28,
        "fontWeight": "500",
        "height": 13.28,
        "lineHeight": 13.28,
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#c3c6cc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 48,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onTextChange={[Function]}
      placeholder=""
      placeholderTextColor="#000000"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 16,
          "height": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <Text
    style={
      Object {
        "color": "#de350b",
        "fontSize": 13.28,
        "fontWeight": "500",
        "height": 13.28,
        "lineHeight": 13.28,
        "marginTop": 8,
      }
    }
  />
</View>
`;

exports[`TextInput renders correctly 2`] = `
<View
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#c3c6cc",
        "fontSize": 13.28,
        "fontWeight": "500",
        "height": 13.28,
        "lineHeight": 13.28,
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#c3c6cc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 48,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onTextChange={[Function]}
      placeholder="Email"
      placeholderTextColor="#000000"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 16,
          "height": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <Text
    style={
      Object {
        "color": "#de350b",
        "fontSize": 13.28,
        "fontWeight": "500",
        "height": 13.28,
        "lineHeight": 13.28,
        "marginTop": 8,
      }
    }
  />
</View>
`;

exports[`TextInput renders correctly 3`] = `
<View
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#c3c6cc",
        "fontSize": 13.28,
        "fontWeight": "500",
        "height": 13.28,
        "lineHeight": 13.28,
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#c3c6cc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 48,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onTextChange={[Function]}
      placeholder="Email"
      placeholderTextColor="#000000"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#000000",
          "flex": 1,
          "fontSize": 16,
          "height": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <View
        override={
          Object {
            "color": "#c3c6cc",
            "height": 24,
            "width": 24,
          }
        }
        xml={
          Object {
            "testUri": "../../../lib/src/icons/email.svg",
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#de350b",
        "fontSize": 13.28,
        "fontWeight": "500",
        "height": 13.28,
        "lineHeight": 13.28,
        "marginTop": 8,
      }
    }
  />
</View>
`;

exports[`TextInput renders correctly 4`] = `
<View
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#c3c6cc",
        "fontSize": 13.28,
        "fontWeight": "500",
        "height": 13.28,
        "lineHeight": 13.28,
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#c3c6cc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 48,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onTextChange={[Function]}
      placeholder="Email"
      placeholderTextColor="#c3c6cc"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#c3c6cc",
          "flex": 1,
          "fontSize": 16,
          "height": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <View
        override={
          Object {
            "color": "#c3c6cc",
            "height": 24,
            "width": 24,
          }
        }
        xml={
          Object {
            "testUri": "../../../lib/src/icons/email.svg",
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#de350b",
        "fontSize": 13.28,
        "fontWeight": "500",
        "height": 13.28,
        "lineHeight": 13.28,
        "marginTop": 8,
      }
    }
  />
</View>
`;

exports[`TextInput renders correctly 5`] = `
<View
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#de350b",
        "fontSize": 13.28,
        "fontWeight": "500",
        "height": 13.28,
        "lineHeight": 13.28,
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#de350b",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 48,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onTextChange={[Function]}
      placeholder="Email"
      placeholderTextColor="#c3c6cc"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#c3c6cc",
          "flex": 1,
          "fontSize": 16,
          "height": 48,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <View
        override={
          Object {
            "color": "#de350b",
            "height": 24,
            "width": 24,
          }
        }
        xml={
          Object {
            "testUri": "../../../lib/src/icons/email.svg",
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#de350b",
        "fontSize": 13.28,
        "fontWeight": "500",
        "height": 13.28,
        "lineHeight": 13.28,
        "marginTop": 8,
      }
    }
  >
    error
  </Text>
</View>
`;
